wandb/sdk/launch/environment/aws_environment.py

```diff
@@ -0,0 +1,216 @@
+"""Implements the AWS environment."""
+
+import os
+import re
+
+from wandb.errors import LaunchError
+from wandb.util import get_module
+
+from .abstract import AbstractEnvironment
+
+boto3 = get_module("boto3", required="AWS environment requires boto3 to be installed.")
+botocore = get_module(
+    "botocore", required="AWS environment requires botocore to be installed."
+)
+
+s3_uri_re = re.compile(r"s3://([^/]+)/(.+)")
+
+
+class AwsEnvironment(AbstractEnvironment):
+    """AWS environment."""
+
+    def __init__(
+        self,
+        region: str,
+        access_key: str,
+        secret_key: str,
+        session_token: str,
+        verify: bool = True,
+    ) -> None:
+        """Initialize the AWS environment.
+
+        Args:
+            region (str): The AWS region.
+
+        Raises:
+            LaunchError: If the AWS environment is not configured correctly.
+        """
+        super().__init__()
+        self._region = region
+        self._access_key = access_key
+        self._secret_key = secret_key
+        self._session_token = session_token
+        if verify:
+            self.verify()
+
+    @classmethod
+    def from_default(cls, verify: bool = True) -> "AwsEnvironment":
+        """Create an AWS environment from the default AWS environment.
+
+        Returns:
+            AwsEnvironment: The AWS environment.
+        """
+        try:
+            session = boto3.Session()
+            region = session.region_name
```

Comment:
I don't believe this will work in the case of an instance role

Reply:
(just the region)

Reply:
I've made the region a required field for the aws environment :)